fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/FlammableBlockRegistry.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.content.registry.v1;
+
+import net.minecraft.block.Block;
+import net.minecraft.tag.Tag;
+
+import net.fabricmc.fabric.api.content.registry.v1.util.ContentRegistry;
+import net.fabricmc.fabric.impl.content.registry.FlammableBlockRegistryImpl;
+
+/**
+ * Registry of Blocks that can be set on fire.
+ */
+public interface FlammableBlockRegistry extends ContentRegistry<Block, FlammableBlockRegistry.Entry> {
+	/**
+	 * Create a new flammable block registry for a type of fire.
+	 * @param fireBlock the block that this registry belongs to
+	 * @param prefix the name of the block
+	 */
+	default FlammableBlockRegistry create(Block fireBlock, String prefix) {
+		return new FlammableBlockRegistryImpl(fireBlock, prefix);
+	}
+
+	default void add(Block block, int burn, int spread) {
+		this.add(block, new Entry(burn, spread));
+	}
+
+	default void add(Tag<Block> tag, int burn, int spread) {
+		this.add(tag, new Entry(burn, spread));
+	}
+
+	final class Entry {
```

Comment:
Entry should be in impl

Reply:
No, because you can add, get, or remove Entries. Making it an inaccessible class doesn't make sense.